transport_nantes/mobilito/views.py

```diff
@@ -166,13 +174,21 @@ class AddressFormView(LoginRequiredMixin, FormView):
     success_url = reverse_lazy('mobilito:recording')
 
     def get_context_data(self, **kwargs):
-        context = super().get_context_data(**kwargs)
+        context = super(FormView, self).get_context_data(**kwargs)
         # Clear session data if user had filled the form before.
         self.request.session['address'] = None
         self.request.session['city'] = None
         self.request.session['postcode'] = None
         return context
 
+    def get(self, request, *args, **kwargs):
+        context = self.get_context_data(**kwargs)
+        if self.get_pages_to_visit():
```

Comment:
This deserves a comment.  I _think_ what you mean is that if the user arrives without having specified a page, you'll 302 (don't 301!) to the correct URL.  But it is not obvious from reading the code why you would do that, even if I know because we've been talking about it.

Reply:
Added a comment, and HttpResponseRedirect has a status code of 302, to get a 301 code it would be a HttpResponsePermanentRedirect, which isn't appropriate here I agree.